From 09669b4515395bdee36be40c1fa0ec77df618750 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Tue, 12 Jan 2016 18:20:13 +0800
Subject: [PATCH 093/666] mmc: sdhci-of-esdhc: T4160/T4080 use the same
 workarounds with T4240

The Freescale QorIQ T4 family has the same eSDHC. So add the same
workarounds with T4240 for T4160/T4080.

Signed-off-by: Li Jingyu <b41874@freescale.com>
Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/mmc/host/sdhci-of-esdhc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-of-esdhc.c b/drivers/mmc/host/sdhci-of-esdhc.c
index 5ebd7f6..1d96ad5 100644
--- a/drivers/mmc/host/sdhci-of-esdhc.c
+++ b/drivers/mmc/host/sdhci-of-esdhc.c
@@ -415,12 +415,13 @@ static void esdhc_of_adma_workaround(struct sdhci_host *host, u32 intmask)
 	 * Check for A-004388: eSDHC DMA might not stop if error
 	 * occurs on system transaction
 	 * Impact list:
-	 * T4240-4160-R1.0 B4860-4420-R1.0-R2.0 P1010-1014-R1.0
+	 * T4240-4160-4080-R1.0 B4860-4420-R1.0-R2.0 P1010-1014-R1.0
 	 * P3041-R1.0-R2.0-R1.1 P2041-2040-R1.0-R1.1-R2.0
 	 * P5020-5010-R2.0-R1.0 P5040-5021-R2.0-R2.1
 	 */
 	if (!(((esdhc->soc_ver == SVR_T4240) && (esdhc->soc_rev == 0x10)) ||
 		((esdhc->soc_ver == SVR_T4160) && (esdhc->soc_rev == 0x10)) ||
+		((esdhc->soc_ver == SVR_T4080) && (esdhc->soc_rev == 0x10)) ||
 		((esdhc->soc_ver == SVR_B4860) && (esdhc->soc_rev == 0x10)) ||
 		((esdhc->soc_ver == SVR_B4860) && (esdhc->soc_rev == 0x20)) ||
 		((esdhc->soc_ver == SVR_B4420) && (esdhc->soc_rev == 0x10)) ||
-- 
2.7.4

